### PR TITLE
Move devel box to example file to force users to configure environment

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
@akofink @stbenjam @jlsherrill @daviddavis @beav 

Our current approach to dev boxes is to give users an environment for development quick and easy. I think it behooves us to encourage long and new users to wind up with an environment that prepares them to make contributions as quickly and easily as possible. This approach while requiring the copying and configuration step pushes users to think about and configure things up front so when the environment is done they are ready to go.

**Comment by beav** [comment]
I am :+1: for this approach, it prevents a new contributor from getting stuck with an error message with no obvious way to continue.

**Comment by beav** [comment]
ACK from me, but I would be interested in comments from others :koala:

File: README.md [hunk 1]
```diff
@@ -65,15 +65,31 @@ This will create a libvirt based virtual machine running the Katello server on C
 
 ### Development Deployment
 
-A Katello development environment can be deployed on CentOS 6 or 7. Ensure that you have followed the steps to setup Vagrant and the libvirt plugin.
+A Katello development environment can be deployed on CentOS 6 or 7. Ensure that you have followed the steps to setup Vagrant and the libvirt plugin. There are a variety of useful development environment options that should or can be set when creating a development box. These options are designed to configure your environment ready to use your own fork, and create pull requests. To create a development box:
 
-To deploy to CentOS 6:
+  1. Copy boxes.yaml.example to boxes.yaml. If you already have a boxes.yaml, you can copy the entries in boxes.yaml.example to your boxes.yaml.
+  2. Now, replace <my_github_username> with your github username
+  3. Fill in any other options, examples:
```
Comment (akofink) [inline README.md]: Maybe say something about the other options available or where to find them, such as [puppet-katello_devel](https://github.com/Katello/puppet-katello_devel/blob/master/manifests/init.pp)?

**Comment by akofink** [comment]
Looks great, @ehelms. It's very similar to other Ruby/Rails projects (i.e. database.yml.example), so I don't feel like it's a huge barrier to entry. Just the one comment from me, otherwise ACK!

**Comment by stbenjam** [comment]
Looks good to me as well, ACK

**Comment by stbenjam** [comment]
Actually, one small thought I just had, what do you think about including another example to show that you can derive from other boxes than base OS boxes?  Maybe a box that inherits from centos7-katello-nightly and flips an installer option (maybe something like turning on rex?)

**Comment by ehelms** [comment]
@stbenjam something more than this section https://github.com/Katello/forklift#adding-custom-boxes ? or improving that section with more examples?

**Comment by stbenjam** [comment]
@ehelms I meant in the boxes.yml.example, but it's not a big deal, you can leave it as is because the documentation is enough (if people read it). Anyway, ACK

**Comment by ehelms** [comment]
@stbenjam on the same page now -- check the updated example file

**Comment by stbenjam** [comment]
Ah ok cool, thanks, looks good :+1:

**Comment by stbenjam** [comment]
I take that back, the _'s don't work in the name of the box.

```
There are errors in the configuration of this machine. Please fix
the following errors and try again:

vm:
* The hostname set for the VM should only contain letters, numbers,
hyphens or dots. It cannot start with a hyphen or dot.
```

**Comment by stbenjam** [comment]
Also anyone that has a box derived from centos7-devel is going to run into problems as it's no longer a base box:

```
/home/stbenjam/git/katello-deploy/Vagrantfile:27:in `fetch': key not found: "box_name" (KeyError)
	from /home/stbenjam/git/katello-deploy/Vagrantfile:27:in `block in define_vm'
	from /usr/share/vagrant/lib/vagrant/config/v2/loader.rb:37:in `call'
```

Maybe send out an e-mail for everyone to merge the new boxes.yml.example into the top of their own.

**Comment by akofink** [comment]
Probably worth a project-wide email blast to let people know, since most of us have a centos7-devel derived box.

**Comment by ehelms** [comment]
@stbenjam updated the example

**Comment by stbenjam** [comment]
ACK

**Pull request merged**